facets_dive/components/facets_dive_vis/facets-dive-vis.ts

```diff
@@ -872,6 +890,48 @@ class FacetsDiveVizInternal {
       selectedData.push(this.elem.data[this.elem.selectedIndices[i]]);
     }
     this.elem.set('selectedData', selectedData);
+    this.updateSelectedBoxes();
+  }
+
+  /**
+   * Update visual appearance of selected items. This code relies heavily on the
+   * D3 join/update/enter/exit pattern.
+   */
+  updateSelectedBoxes() {
+    const selectedBoxes: ItemPosition[] =
+      this.elem.selectedIndices.map(index => {
+        return {x: this.spriteMesh.getX(index), y: this.spriteMesh.getY(index)};
+    });
+
+    // JOIN.
+    const selectedElements = this.selectedLayer.selectAll('.selected').data(
+      selectedBoxes);
+
+    selectedElements
+        // ENTER.
+        .enter()
+        .append('rect')
+        .attr('class', 'selected')
+        .attr('x', (pos: ItemPosition) => pos.x || 0)
+        .attr('y', (pos: ItemPosition) => pos.y || 0)
+        .attr('width', 1)
+        .attr('height', 1)
+        .attr('stroke', SELECTED_ITEM_COLOR)
+        .attr('stroke-opacity', 0)
+        .attr('stroke-width', SELECTED_ITEM_STROKE_WIDTH)
+        .attr('fill-opacity', 0)
+        // ENTER + UPDATE.
+        .merge(selectedElements)
+        .attr('x', (pos: ItemPosition) => pos.x || 0)
+        .attr('y', (pos: ItemPosition) => pos.y || 0)
+        .attr('width', 1)
+        .attr('height', 1)
+        .attr('stroke-opacity', 1);
+
+    // EXIT.
+    selectedElements.exit()
+        .remove()
+        .attr('stoke-opacity', 0);
```

Comment:
Setting stroke-opacity to 0 for a removed element should have no effect. If you wanted to fade the elements out smoothly, you'd do that with a transition before the remove.

~~~
selectedElements.exit().transition().attr('stroke-opacity', 0).remove();
~~~

But since the rest of this doesn't use a transition, there's no need to do this here.

Reply:
fixed